### PR TITLE
fix: 🐛 don't add empty files to bundles [ROAD-612]

### PR DESCRIPTION
**Pull request by bastiandoetsch**

### CODE REVIEW

File: src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java [hunk 1]
```diff
@@ -351,10 +351,17 @@ private boolean isNotSucceed(
           progress, PREPARE_FILES_TEXT + fileCounter + " of " + totalFiles + " files done.");
 
       final String path = pdUtils.getDeepCodedFilePath(file);
+
+      // don't analyse empty files
+      if (pdUtils.getFileSize(file) == 0) {
```
Comment (ArtsiomCh) [inline src/main/java/ai/deepcode/javaclient/core/AnalysisDataBase.java:356]: Seems better place for that check will be `DeepCodeUtilsBase#isSupportedFileFormat`
Then we'll have correct files at `getAllSupportedFilesInProject` too.

File: src/main/java/ai/deepcode/javaclient/core/DeepCodeUtilsBase.java [hunk 2]
```diff
@@ -77,12 +71,10 @@ protected DeepCodeUtilsBase(
...
     if (ignoreInfoHolder.isIgnoredFile(file) || isGitIgnoredExternalCheck(file)) return false;
-    final boolean result =
-        getFileLength(file) < MAX_FILE_SIZE
-            && (supportedExtensions.contains(getFileExtention(file))
-                || supportedConfigFiles.contains(pdUtils.getFileName(file)));
     // DCLogger.getInstance().info("isSupportedFileFormat ends for " + psiFile.getName());
-    return result;
+    return getFileLength(file) < MAX_FILE_SIZE
+      && (supportedExtensions.contains(getFileExtention(file)) || supportedConfigFiles.contains(pdUtils.getFileName(file)))
+      && pdUtils.getFileSize(file) > 0;
```
Comment (ArtsiomCh) [inline src/main/java/ai/deepcode/javaclient/core/DeepCodeUtilsBase.java:77]: I mean to create once:
`long fileLength = getFileLength(file)`
And then do `0 < fileLength && fileLength < MAX_FILE_SIZE`

File: src/main/java/ai/deepcode/javaclient/core/DeepCodeUtilsBase.java [hunk 3]
```diff
@@ -77,12 +71,10 @@ protected DeepCodeUtilsBase(
   public boolean isSupportedFileFormat(@NotNull Object file) {
     // DCLogger.getInstance().info("isSupportedFileFormat started for " + psiFile.getName());
     if (ignoreInfoHolder.isIgnoredFile(file) || isGitIgnoredExternalCheck(file)) return false;
-    final boolean result =
-        getFileLength(file) < MAX_FILE_SIZE
-            && (supportedExtensions.contains(getFileExtention(file))
-                || supportedConfigFiles.contains(pdUtils.getFileName(file)));
     // DCLogger.getInstance().info("isSupportedFileFormat ends for " + psiFile.getName());
```
Comment (ArtsiomCh) [inline src/main/java/ai/deepcode/javaclient/core/DeepCodeUtilsBase.java:74]: That line has no sense here after moving all condition checks into return.
Would be useful for debug to have it separated as it was before.

**Comment by bastiandoetsch** [comment]
Let's release today, we need the fix in the intellij plugin asap :)

**Review by ArtsiomCh — APPROVED** [review]
LGTM

**Pull request merged**